modules/web/services/darkhttpd/default.nix

```diff
@@ -0,0 +1,206 @@
+we don't know if the server needs temporary files in any way and by using a
+separate temporary directory we protect from any potential data leaks resulting
+from that.
+
+There are some fancier options one can use, such as:
+
+    foo.serviceOwner = "roman";
+    foo.proxyOptions.TLS = "none";
+    foo.proxyOptions.http.mode = "on";
+    foo.proxyOptions.https.mode = "off";
+    foo.proxyOptions.path = "/"; };
+
+'';
+
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ leenaars ];
+    meta.platforms = lib.platforms.linux;
+  };
+
+   config = let
+    serviceCommand = "${pkgs.${config.serviceName}}/bin/${serviceName}";
+    serviceParameters = [
+      "${toString config.staticDir}"
+      "--addr" "127.0.0.1"
+      "--port" "${toString config.proxyOptions.port}"
+      "--no-server-id"
+      "--daemon"
+      "--no-keepalive"
+      "--mimetypes" "${pkgs.apacheHttpd}/conf/mime.types"
+    ] ++ optional (!config.directoryListing) "--no-listing"
+      ++ optional (config.log) "--log ${config.logFile}";
+  in mkIf config.enable rec {
+
+# TODO WISHLIST: if the user flags that he wants to work with a fixed copy of
+# the site a clone should be made. For tiny static sites, that could even be in
+# the Nix store. This would allow rollbacks etc.
+
+# TODO WISHLIST: We assume that the owner of domain name example.com may need
+# to have access to all the assets from that domain, so that different services
+# may be combined. For instance, a remote text editor instance such as Leaps
+# without the strict access restriction we apply to this instance could be used
+# to manually edit some static content served on a public facing website.
+# Therefore group ownership is set to the value of serviceOwner as configured
+# by the user. The user can also choose to make this a specific group of
+# course.
+
+# TODO WISHLIST: Allow a remote URI (e.g. git/darcs/mercurial repository or ZIP
+# file) to be used; in case of a repository this could be polled at some
+# interval chosen.
+
+    groups."${config.serviceOwner}" = {};
+
+    users."${config.serviceName}" = {
+      description = "${config.serviceName} server user for ${config.serviceOwner}";
+      home        = config.staticDir;
+      createHome  = false;
+      group = config.serviceOwner;
+    };
+
+    systemd.services."${config.serviceName}" = {
+      description   = "${config.serviceName} service created by Nixcloud";
+      wantedBy      = [ "multi-user.target" ];
+      after         = [ "network.target" ];
+
+      serviceConfig = {
+        User = "${config.serviceName}";
+        Group = "${config.serviceOwner}";
+        Restart = "always";
+        WorkingDirectory = config.stateDir;
+	MemoryDenyWriteExecute = true;
+	RestrictNameSpaces = true;
+	NoNewPrivileges = true;
+	ProtectHome = true;
+	PrivateUsers = true;
+	ProtectSystem = true;
+        PrivateTmp = true;
+	ProtectKernelTunables = true;
+	LogsDirectory = config.logFile;
+	Type = "forking";
```

Comment:
I'd remove the `--daemon` argument from the arguments and remove the `Type` here as well (so it defaults to `simple`), because systemd *already* forks off, so no need to do it another time.